src/libs/Middleware/RecheckConnection.ts

```diff
@@ -1,20 +1,17 @@
 import CONST from '../../CONST';
 import NetworkConnection from '../NetworkConnection';
+import Middleware from './types';
 
 /**
- * @returns {Function} cancel timer
+ * @returns cancel timer
  */
-function startRecheckTimeoutTimer() {
+function startRecheckTimeoutTimer(): () => void {
     // If request is still in processing after this time, we might be offline
     const timerID = setTimeout(NetworkConnection.recheckNetworkConnection, CONST.NETWORK.MAX_PENDING_TIME_MS);
     return () => clearTimeout(timerID);
 }
 
-/**
- * @param {Promise} response
- * @returns {Promise}
- */
-function RecheckConnection(response) {
+const RecheckConnection: Middleware = (response) => {
```

Comment:
We should add a type to `Response`

Reply:
Same as https://github.com/Expensify/App/pull/28031#discussion_r1345580714